core/tests/test_utils.py

```diff
@@ -640,7 +640,8 @@ def mark_user_for_deletion(self, user_id):
                 deleted.
         """
         self._user_id_by_auth_id = {
-            a: u for a, u in self._user_id_by_auth_id.items() if u != user_id
+            a: u if u['id'] != user_id else {'id': u['id'], 'deleted': True}
```

Comment:
This seems a bit weird to me. Why the change in logic?

It might also be better to do stuff using domain objects here, instead. I can't tell if `u` has the same structure as the dict in the else clause. (Similarly below.)

Reply:
Since there was a skew between the stubs and the library code. Yeah, domain objects might be better.

Reply:
I have changed this and also added better names to the variables.